File: python-examples/cdp-connection/hooks/setup_context.py
from intuned_runtime import attempt_store


async def setup_context(*, api_name: str, api_parameters: str, cdp_url: str):
    """
    Setup hook that runs before API execution.
    Captures the CDP URL from Intuned's runtime and stores it in attempt_store.
    """
    attempt_store.set("cdp_url", cdp_url)

File: typescript-examples/cdp-connection/hooks/setupContext.ts
import { attemptStore } from "@intuned/runtime";

export default async function setupContext({
  cdpUrl,
  apiName,
  apiParameters,
}: {
  cdpUrl: string;
  apiName: string;
  apiParameters: any;
}) {
  attemptStore.set("cdpUrl", cdpUrl);
}
